### PR TITLE
Update after recent rust-nightly changes.

### PR DESCRIPTION
**Pull request by ltratt**
There are two PRs which affect us:

  https://github.com/rust-lang/rust/pull/74850
  https://github.com/rust-lang/rust/pull/75152

Henceforth, we should probably be keeping a close eye on and contributing to:

  https://github.com/rust-lang/wg-allocators/issues

### CODE REVIEW

**Comment by jacob-hughes** [comment]
Good idea. This stuff is moving quite fast and breakages tend to happen a lot in the alloc API.

bors r+

**Comment by ltratt** [comment]
Sausages. We've been hoist by rustfmt's petard: the last version of rustc with a working rustfmt predates this change :/

Any thoughts?

**Comment by ltratt** [comment]
Thinking about this, I think that we have to wait to merge this PR, because we have dependent repos that expect a working rustfmt.

What that means is that, when building locally, one needs to use a specific version of rust (2020-07-27 to be specific) until rustfmt builds again and we can merge this. What a mess!

**Comment by ltratt** [comment]
bors try

**Comment by ltratt** [comment]
bors try

**Comment by ltratt** [comment]
bors try

**Comment by ltratt** [comment]
bors try

**Comment by ltratt** [comment]
bors try

**Comment by ltratt** [comment]
bors try

**Comment by ltratt** [comment]
OK for me to squash?

**Comment by ltratt** [comment]
bors try

**Comment by ltratt** [comment]
bors try

File: .buildbot.sh [hunk 1]
```diff
@@ -6,11 +6,10 @@ export CARGO_HOME="`pwd`/.cargo"
 export RUSTUP_HOME="`pwd`/.rustup"
 
 curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs > rustup.sh
-# Install no toolchain initially to ensure toolchain rollback.
-sh rustup.sh --default-host x86_64-unknown-linux-gnu --default-toolchain none -y --no-modify-path
-
+sh rustup.sh --default-host x86_64-unknown-linux-gnu --default-toolchain nightly -y --no-modify-path
```
Comment (vext01) [inline .buildbot.sh]: can we select the minimal profile here? If so we can kill the following invocation of rustup?

**Comment by ltratt** [comment]
bors try

File: .buildbot.sh (hunk 1 above)
Comment (ltratt) [reply .buildbot.sh]: We can. Fixed in https://github.com/softdevteam/rboehm/pull/19/commits/0a4753601def7808deb3a6292c56f1e807eee195.

**Comment by ltratt** [comment]
I found the number of args was getting out of hand, so I've done a reformat. But otherwise I think this might be ready for squashing?

**Comment by vext01** [comment]
Great!

Can't comment on the other changes, but looks like it's pre-reviewed by Jake.

Please squash.

**Comment by ltratt** [comment]
Squashed.

**Pull request merged**